scripts/validate_docstrings.py

```diff
@@ -344,30 +338,36 @@ def get_all_api_items():
 
 
 def print_validate_all_results(
-    prefix: str,
-    errors: list[str] | None,
     output_format: str,
+    prefix: str | None,
+    errors: list[str] | None,
     ignore_deprecated: bool,
-    ignore_functions: list[str] | None,
+    for_error_ignore_functions: dict[str, list[str]] | None,
 ):
     if output_format not in ("default", "json", "actions"):
         raise ValueError(f'Unknown output_format "{output_format}"')
+    if for_error_ignore_functions is None:
+        for_error_ignore_functions = {}
 
-    result = validate_all(prefix, ignore_deprecated, ignore_functions)
+    result = validate_all(prefix, ignore_deprecated)
 
     if output_format == "json":
         sys.stdout.write(json.dumps(result))
         return 0
 
     prefix = "##[error]" if output_format == "actions" else ""
     exit_status = 0
-    for name, res in result.items():
+    for func_name, res in result.items():
         for err_code, err_desc in res["errors"]:
+            ignore_functions = for_error_ignore_functions.get(err_code, [])
             if errors and err_code not in errors:
                 continue
+            elif func_name in ignore_functions:
```

Comment:
I'd implement a function and write something like:
```python
if is_error_ignored(func_name, ignore_errors):
    continue
```

There conditions are a bit tricky to understand, encapsulating them in a function will make code clearer in my opinion.

Reply:
Done, though instead I introduced some variables instead of redirecting to a different function. I found that simpler than redirecting to another function.